lib/libc/minimal/CMakeLists.txt

```diff
@@ -35,6 +35,10 @@ zephyr_library_sources(
   ${STRERROR_TABLE_H}
 )
 
+zephyr_library_sources_ifdef(CONFIG_POSIX_FILE_SYSTEM
+  source/file_ops.c
+  )
+
```

Comment:
`fclose()` should be implemented by the C library. It should already work in a number of cases.

Additionally, functions that are part of the `POSIX_FILE_SYSTEM` Option Group (which are not a part of the C library) should be implemented in the file `fs.c`, I believe (although we may eventually want to rename `fs.c` to `file_system.c` to make that relationship more clear).

```suggestion
```

Reply:
I am little bit confused, but `fopen` is implemented in minimal libc already, similar to sqrt and other functions.

For the dependency on POSIX, what I can think of is that `fopen`/`fclose` has to depend on `_open`/`_close`, which will depend on POSIX implementation if `COFIG_POSIX_FILE_SYSTEM` is true. Is that what you mean ?

Reply:
POSIX is optional in Zephyr whereas C is not. It is possible to use plain C without linking to the POSIX API.

Additionally, both C and POSIX can be layered on top of the same Zephyr call (a mutual dependency).